share/lib/hoc/import3d/read_swc.hoc

```diff
@@ -292,36 +316,59 @@ proc mksections() {local i, j, isec, first  localobj sec, psec, pts
 	mksection(isec, first, i)
 }
 
-proc mksection() { local i, isec, first   localobj sec
-	isec = $1  first=$2  i=$3
-	if (isec > 0) {// branches have pid as first point
-		sec = new Import3d_Section(first, i-first+1)
-		pt2sec(pix2ix(first), sec.parentsec)
-		// but if the parent is the root and the branch has more than
-		// one point, then connect to center of root with wire
-		if (point2sec.x[pix2ix(first)] == 0 && i > 1) {
-			sec.parentx = 0.5
-			sec.first = 1
-		}else{
-			if (pix2ix(first) == 0) { sec.parentx = 0 }
-		}
-		sec.append(0, pix2ix(first), 1, x, y, z, d)
-		sec.append(1, first, i-first, x, y, z, d)
-	}else{// pid not first point in root section
-		sec = new Import3d_Section(first, i-first)
-		sec.append(0, first, i-first, x, y, z, d)
-	}
-	sec.type = type.x[first]
-	sections.append(sec)
-	if (object_id(sec.parentsec)) {
-		if (sec.parentsec.type == 1 && sec.type != 1) {
-			sec.d.x[0] = sec.d.x[1]
-		}
-	}
-	if (connect2prox.x[first]) {
-		sec.pid = sec.parentsec.id
-		sec.parentx = 0
-	}
+proc mksection() { local i, isec, first, den_con_soma   localobj sec, psec
+  isec = $1  first=$2  i=$3
+  if (isec == 0) {// Root. pid not first point in root section
+    sec = new Import3d_Section(first, i-first)
+    sec.append(0, first, i-first, x, y, z, d)
+  }else{ // Not root. Branches have pid as first point
+    sec = new Import3d_Section(first, i-first+1)
+    pt2sec(pix2ix(first), sec.parentsec)
+    psec = sec.parentsec
+
+    // 1) If 1 point soma, all branches connect with wire to parentx=0.5;
+    // 2) All branches that connect to last point of parent have parentx=1;
+    // 3) All branches that connect to first point of root, then parentx=0;
+
+    // 4) If branch is not soma and connects to interior of type 1 root, use
+    //    pt3dtype of 1 and parentx=0.5 (best if parent point is middle point)
```

Comment:
Why always attach to the middle?

Some somas have many points, and we could conceivably attach to any of them. We know the arc-length, so we know the normalized x.

Reply:
The implicit assumption is that the soma is fat and needs only a single compartment (branching somas will appear as arrays). Connecting to 0.5 with a wire is a convenience. CellBuild I think only knows about 0, .5, and 1.0 connections and
.5  and 0 is only allowed for the root.  If one does not use the soma type (1) then every branch means the parent comes to an end at that point.  But I admit that counter arguments have force.